shell/platform/fuchsia/dart_runner/dart_runner.cc

```diff
@@ -152,13 +156,112 @@ void RunApplicationV2(
   }
 }
 
+void RunTestApplicationV2(
+    DartRunner* runner,
+    fuchsia::component::runner::ComponentStartInfo start_info,
+    std::shared_ptr<sys::ServiceDirectory> runner_incoming_services,
+    fidl::InterfaceRequest<fuchsia::component::runner::ComponentController>
+        controller,
+    fit::function<void(std::shared_ptr<DartTestComponentControllerV2>)>
+        component_created_callback,
+    fit::function<void(DartTestComponentControllerV2*)> done_callback) {
+  const int64_t start = Dart_TimelineGetMicros();
+
+  auto test_component = std::make_shared<DartTestComponentControllerV2>(
+      std::move(start_info), runner_incoming_services, std::move(controller),
+      std::move(done_callback));
+
+  component_created_callback(test_component);
+
+  // Start up the dart isolate and serve the suite protocol.
+  test_component->SetUp();
+
+  const int64_t end = Dart_TimelineGetMicros();
+  Dart_TimelineEvent("DartTestComponentControllerV2::SetUp", start, end,
+                     Dart_Timeline_Event_Duration, 0, NULL, NULL);
+}
+
 bool EntropySource(uint8_t* buffer, intptr_t count) {
   zx_cprng_draw(buffer, count);
   return true;
 }
 
 }  // namespace
 
+// "data" and "assets" are arguments that are specific to the Flutter runner.
+// They will likely go away if we migrate to the ELF runner.
+constexpr char kDataKey[] = "data";
+constexpr char kAssetsKey[] = "assets";
+
+// "args" are how the component specifies arguments to the runner.
+constexpr char kArgsKey[] = "args";
+constexpr char kOldGenHeapSizeKey[] = "old_gen_heap_size";
+constexpr char kExposeDirsKey[] = "expose_dirs";
+
+/// Parses the |args| field from the "program" field into
+/// |metadata|.
+void ParseArgs(std::vector<std::string>& args, ProgramMetadata* metadata) {
+  // fml::CommandLine expects the first argument to be the name of the program,
+  // so we prepend a dummy argument so we can use fml::CommandLine to parse the
+  // arguments for us.
+  std::vector<std::string> command_line_args = {""};
+  command_line_args.insert(command_line_args.end(), args.begin(), args.end());
+  fml::CommandLine parsed_args = fml::CommandLineFromIterators(
+      command_line_args.begin(), command_line_args.end());
+
+  std::string is_test;
+  if (parsed_args.GetOptionValue("is_test", &is_test)) {
```

Comment:
We don't seem to use any of this ProgramMetadata besides is_test. Should we remove the rest of the logic and simplify this down to:

```c++
bool IsTestProgram(
    const fuchsia::data::Dictionary& program_metadata) {
  for (const auto& entry : program_metadata.entries()) {
    if (entry.key.compare(kArgsKey) != 0 || entry.value == nullptr) {
      continue;
    }
    auto args = entry.value->str_vec();

    // Comment omitted for brevity.
    std::vector<std::string> command_line_args = {""};
    command_line_args.insert(command_line_args.end(), args.begin(), args.end());
    fml::CommandLine parsed_args = fml::CommandLineFromIterators(
        command_line_args.begin(), command_line_args.end());

    std::string is_test_str;
    return parsed_args.GetOptionValue("is_test", &is_test_str) &&
        is_test_str == "true";
  }
  return false;
}
```

and get rid of the `ProgramMetadata` struct from this runner?